src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((result, item) => {
+    const keyValue = item[key] as string;
+
+    const newResult = { ...result };
+
+    if (!newResult[keyValue]) {
+      newResult[keyValue] = [];
+    }
+
+    newResult[keyValue].push(item);
+
+    return newResult;
```

Comment:
Creating a new object on each iteration with `{ ...result }` is inefficient because it copies all properties from the current result to a new object. This can lead to performance issues with a large number of items. It's better to mutate the `result` object directly in this case, as it's already a new object created by the `reduce` function and not shared elsewhere.